code/game/mecha/equipment/tools/janitor_tools.dm

```diff
@@ -0,0 +1,377 @@
+	if(istype(target, /obj/machinery/light))
+		chassis.Beam(target, icon_state = "rped_upgrade", icon = 'icons/effects/effects.dmi', time = 5)
+		playsound(src, 'sound/items/pshoom.ogg', 40, 1)
+		ReplaceLight(target)
+
+/obj/item/mecha_parts/mecha_equipment/janitor/light_replacer/proc/ReplaceLight(obj/machinery/light/target)
+	if(target.status != LIGHT_OK)
+		to_chat(chassis.occupant, "<span class='notice'>You replace the light [target.fitting] with [src].</span>")
+		var/obj/item/light/replacement = target.light_type
+		target.status = LIGHT_OK
+		target.switchcount = 0
+		target.rigged = emagged
+		target.brightness_range = initial(replacement.brightness_range)
+		target.brightness_power = initial(replacement.brightness_power)
+		target.brightness_color = initial(replacement.brightness_color)
+		target.on = target.has_power()
+		target.update(TRUE, TRUE, FALSE)
+	else
+		to_chat(chassis.occupant, "<span class='warning'>There is a working [target.fitting] already inserted!</span>")
+		return
+
+#undef LIGHT_OK
+#undef LIGHT_EMPTY
+#undef LIGHT_BROKEN
+#undef LIGHT_BURNED
+
+// Mecha spray
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_spray
+	name = "JS-33 super spray"
+	desc = "A spray bottle, upscaled for an exosuit. Capable of mass sanitation."
+	icon_state = "mecha_spray"
+	equip_cooldown = 15
+	energy_drain = 200
+	range = MECHA_MELEE | MECHA_RANGED
+	/// Toggle for refilling itself
+	var/refill_enabled = TRUE
+	/// Rate per process() tick spray refills itself
+	var/refill_rate = 1
+	/// Power use per process to refill reagents
+	var/refill_cost = 25
+	/// What reagent to refill with
+	var/refill_reagent = "cleaner"
+	/// The range of tiles the sprayer will reach.
+	var/spray_range = 4
+
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_spray/New()
+	create_reagents(100)
+	reagents.add_reagent("cleaner", 100)
+	START_PROCESSING(SSobj, src)
+	..()
+
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_spray/emag_act(mob/user)
+	. = ..()
+	emagged = TRUE
+	to_chat(user, "<span class='notice'>You short out the automatic watering system on [src].</span>")
+	reagents.clear_reagents()
+	refill_reagent = "lube"
+	refill_cost = 50
+	refill_rate = 5
+
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_spray/action(atom/target)
+	if(reagents.total_volume < 15) // Needs at least enough reagents to apply the full spray
+		return
+	var/direction = get_dir(chassis, target)
+	var/turf/T = get_turf(target)
+	var/turf/T1 = get_step(T, turn(direction, 90))
+	var/turf/T2 = get_step(T, turn(direction, -90))
+	var/list/the_targets = list(T, T1, T2)
+	playsound(chassis, 'sound/effects/spray2.ogg', 75, TRUE, -3)
+	for(var/turf/target_turf in the_targets)
+		INVOKE_ASYNC(src, PROC_REF(spray), target_turf)
+
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_spray/proc/spray(turf/target)
+	spawn(0)
+		var/obj/effect/decal/chempuff/spray = new /obj/effect/decal/chempuff(get_turf(chassis))
+		if(!spray)
+			return
```

Comment:
Unless DM is being super weird (which is always a possibility), there's no way this could not exist here, since you just created it.